pystac_client/cli.py

```diff
@@ -193,6 +198,14 @@ def parse_args(args: List[str]) -> Dict[str, Any]:
         if "json" in parsed_args["filter_lang"]:
             parsed_args["filter"] = json.loads(parsed_args["filter"])
 
+    if "q" in parsed_args:
+        logger.warning("Argument -q is deprecated, use --query instead")
+        if "query" not in parsed_args:
+            parsed_args["query"] = parsed_args["q"]
+        else:
+            logger.error("Both -q and --query arguments specified, ignoring -q")
```

Comment:
Hm, it seems strange to log an error but continue execution. Maybe `warning`?

Reply:
I did error b/c I think it's unlikely that a user would intentionally mix these. Honestly, I think there's no chance that any user is ever going to encounter this, but if they do, I want to make it clear what the behavior is.